### PR TITLE
added solution to the class StorageImpl:

### PR DESCRIPTION
**Pull request by nikolasdymchenko7**
- added implementation to the exists methods: put(), get(), size()
- added two another methods: addToArrays(), is exists(); the first one adds data to the arrays and another one checks about existing values in the array

### CODE REVIEW

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 1]
```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int START_INDEX = 0;
+    private static final int SIZE = 5;
+    private Object[] keys = new Object[SIZE];
+    private Object[] values = new Object[SIZE];
+    private int arrayIndex = 0;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:10]:
```suggestion
    private int size;
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 2]
```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int START_INDEX = 0;
+    private static final int SIZE = 5;
+    private Object[] keys = new Object[SIZE];
+    private Object[] values = new Object[SIZE];
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:9]: image

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 3]
```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int START_INDEX = 0;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:6]:
```suggestion
```

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 4]
```diff
@@ -3,17 +3,59 @@
...
+    private static final int SIZE = 5;
```
Comment (Rommelua) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:7]:
```suggestion
    private static final int MAX_CAPACITY= 10;
```

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/core/basesyntax/impl/StorageImpl.java (hunk 1 above)
Comment (nikolasdymchenko7) [reply src/main/java/core/basesyntax/impl/StorageImpl.java:10]: Got it, thanks

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 5]
```diff
@@ -3,17 +3,61 @@
...
+
+    private boolean isExist(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == key) {
+                addToArrays(key, value, i);
+                return true;
+            } else if (keys[i] != null && keys[i].equals(key)) {
+                addToArrays(key, value, i);
+                return true;
+            }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:42]: the same body - lets combine conditions

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 6]
```diff
@@ -3,17 +3,61 @@
...
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == key) {
+                return (V) values[i];
+            } else if (keys[i] != null && keys[i].equals(key)) {
+                return (V) values[i];
+            }
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:54]: the same body - combine conditions

File: src/main/java/core/basesyntax/impl/StorageImpl.java [hunk 7]
```diff
@@ -3,17 +3,61 @@
...
+    public StorageImpl() {
+        keys = new Object[MAX_CAPACITY];
+        values = new Object[MAX_CAPACITY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size < MAX_CAPACITY) {
+            if (!isExist(key, value)) {
```
Comment (aaananas) [inline src/main/java/core/basesyntax/impl/StorageImpl.java:20]: method isExist is confusing, i dont understand why it adds values

**Review by aaananas — CHANGES_REQUESTED** [review]

**Pull request closed without merging**